sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -342,11 +352,33 @@ def from_estimator(
         else:
             X_grid = np.c_[xx0.ravel(), xx1.ravel()]
 
-        pred_func = _check_boundary_response_method(estimator, response_method)
-        response = pred_func(X_grid)
+        prediction_method = _check_boundary_response_method(
+            estimator, response_method, class_of_interest
+        )
+        try:
+            response, _, response_method_used = _get_response_values(
+                estimator,
+                X_grid,
+                response_method=prediction_method,
+                pos_label=class_of_interest,
+                return_response_method_used=True,
+            )
+        except ValueError as exc:
+            if "is not a valid label" in str(exc):
+                # re-raise a more informative error message since `pos_label` is unknown
+                # to our user when interacting with
+                # `DecisionBoundaryDisplay.from_estimator`
+                raise ValueError(
+                    f"class_of_interest={class_of_interest} is not a valid label: It "
+                    f"should be one of {estimator.classes_}"
+                ) from exc
+            raise
+        except AttributeError as exc:
+            # re-raise the AttributeError as a ValueError for backward compatibility
+            raise ValueError(str(exc)) from exc
```

Comment:
this seems bad to me. Attribute error should be an attribute error, not converted to a ValueError. I would just change this. These displays are used in interactive mode, the type of exception could be changed here IMO.

Reply:
Both `ValueError` and `AttributeError` are valid. `AttributeError` is valid since the estimator does not expose the attribute but the `ValueError` is valid because this is a parameter value passed by the user.

Since we are already raising a `ValueError` that is not semantically wrong, I agree with @ogrisel that this is a pity to break the backward compatibility.

Reply:
I dislike the kind of code which is only there because somebody did something in the past and it's there for historical reasons. Makes code less maintainable over time. And I don't consider this a major backward compatibility issue since it's not like we were working and we don't now. We were failing, and we still fail, but a different error type. If you really want, we can add a message here that the type of the error will be change to an `AttributeError` in 1.6, but I don't think we should keep such code in the long term in the code base.

Reply:
So let's take the second option of the @ogrisel comment: https://github.com/scikit-learn/scikit-learn/pull/27291#discussion_r1334181433

Let's acknowledge the change in the changelog only.

Reply:
I buy the fact that the display are usually used in a more interactive mode and this is I don't foresee a use case where someone will catch the error to do something else.